modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -4629,3 +4637,166 @@ def from_dataframe(cls, df: "ProtocolDataframe") -> "PandasDataframe":
+        wrap_callable = Engine.get() != "Dask"
+        use_map = wrap_callable
+        new_caselist = []
+        for condition, replacement in caselist:
+            if callable(condition):
+                if wrap_callable:
+                    condition = wrapper_put(condition)
+            else:
+                use_map = False
+                if isinstance(condition, cls):
+                    condition = copartition(condition, True)
+            if callable(replacement):
+                if wrap_callable:
+                    replacement = wrapper_put(replacement)
+            elif use_map and is_list_like(replacement):
+                use_map = False
+                if isinstance(replacement, cls):
+                    replacement = copartition(replacement, None)
+            new_caselist.append((condition, replacement))
+
+        # If all the conditions are callable and the replacements are either
+        # callable or scalar, use map().
+        if use_map:
+            return self.map(func=remote_fn, func_args=[new_caselist], lazy=True)
+
+        def map_data(
+            part_offset,
+            part_len,
+            data,
+            data_offset,
+            fill_value,
+        ):
+            if isinstance(data, cls):
+                if part_offset < len(data._partitions):
+                    return data._partitions[part_offset][0]._data
+                else:
+                    return [fill_value] * part_len
+
+            if isinstance(data, pandas.Series):
+                nonlocal is_trivial_idx
+                if is_trivial_idx is None:
+                    is_trivial_idx = is_trivial_index(self.index)
+                if is_trivial_idx and is_trivial_index(data.index):
+                    data = data[data_offset : data_offset + part_len]
+                    diff = part_len - len(data)
+                    if diff > 0:
+                        data = pandas.concat((data, pandas.Series([fill_value] * diff)))
+                    return data
+                return data.reindex(
+                    self.index[data_offset : data_offset + part_len],
+                    fill_value=fill_value,
+                )
+
+            if not wrap_callable and callable(data):
+                return wrapper_put(data)
+
+            return (
+                data[data_offset : data_offset + part_len]
+                if is_list_like(data)
+                else data
+            )
+
+        if parts is None:
+            parts = [p[0] for p in self._partitions]
+            lengths = self._get_lengths(parts, Axis.ROW_WISE)
+        new_parts = []
+        data_offset = 0
+        for i in range(0, parts_len):
+            part = parts[i]
+            part_len = lengths[i]
+            cases = [
+                tuple(
+                    map_data(i, part_len, d[0], data_offset, d[1])
+                    for d in zip(c, (True, None))
```

Comment:
What is the purpose of None?

Reply:
These are the fill values - True fro condition and None for replacement.